server/command.go

```diff
@@ -209,6 +209,15 @@ func (p *Plugin) ExecuteCommand(_ *plugin.Context, args *model.CommandArgs) (*mo
 		return &model.CommandResponse{}, nil
 	}
 
+	if action == commandTriggerSetChannelWelcome || action == commandTriggerGetChannelWelcome || action == commandTriggerDeleteChannelWelcome {
+		hasPermissionTo := p.API.HasPermissionToChannel(args.UserId, args.ChannelId, model.PERMISSION_MANAGE_SLASH_COMMANDS)
+
+		if !hasPermissionTo {
+			p.postCommandResponse(args, "/%s commands can only be executed by the user with permissions to manage the slash commands", action)
+			return &model.CommandResponse{}, nil
+		}
+	}
+
 	isSysadmin, err := p.hasSysadminRole(args.UserId)
```

Comment:
We are checking if the user is a sysadmin afterwards here, even though we have already vetted the user for some commands not related to being a sysadmin. Can we have the sysadmin check for sysadmin-specific commands first, then check for channel admin for channel admin specific commands?

Also, the command "help text" should be updated to note which commands require channel admin permissions versus sysadmin permissions.

Reply:
@mickmister  got it. I'll revert the steps.